File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        mx = max(candies) - extraCandies
        arr = []
        for i in candies:
            if i >= mx:
                arr.append(True)
            else:
                arr.append(False)
                
        return arr

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int max = 0;
        for(int i = 0 ; i  < candies.length; i++){
            if (candies[i] > max){
                max = candies[i];
            }
        }
        max = max - extraCandies;
        List<Boolean> arr = new ArrayList<Boolean>();
        
        for(int i = 0 ; i  < candies.length; i++){
            if (candies[i] >= max){
                arr.add(true);
            }
            else{
                arr.add(false);
            }
        }
        return arr;
    }
}